sklearn/neighbors/base.py

```diff
@@ -759,6 +759,11 @@ def fit(self, X, y):
 
         y : {array-like, sparse matrix}
             Target values of shape = [n_samples] or [n_samples, n_outputs]
+            If y is sparse with values [0, 1], it is expected to be the
+            indicator representation of a multilabel classification problem.
+            Otherwise, if y is an array with integer dtype, it is interpreted
```

Comment:
y can have non-integer dtype in the multiclass case.

Better to just say that "A sparse matrix may only be used for multilabel targets."